### DISCUSSION TITLE
Create React App + Compiled

### DISCUSSION THREAD

**Discussion by DanBlakeNZ** [atlassian-labs/compiled#846] [Q&A]
Hi there, 
Our team is interested in using Complied and I am wondering how well it works (if at all) with Create React App currently? I have got Complied working with a manual Babel, Webpack & React app and it works quite nicely. But when trying to add  Compiled to a CRA application we obviously can't access webpack config without ejecting or using something like 'Create React App Configuration Override (CRACO)'. I have tried using CRACO and adding Compiled to craco.config.js (ie webpack.config) file, but then started running into errors from Compiled `You haven't configured a transformer yet` etc.
So before going any further, I was just wondering  what does the process look like in terms of getting it up and running on a CRA application - or does it require a whole lot of extra configuration?

Thanks!

**Comment by MonicaOlejniczak** [comment]
Unfortunately compiled is not compatible with CRA, unless you eject the application or override the configuration, like you have said. Though when you do that, the installation should be very similar to what is already outlined on the [website](https://compiledcssinjs.com/docs/installation) using either babel or webpack.

In the future, compiled should expose a babel macro like [emotion](https://emotion.sh/docs/babel-macros) does. That would enable anyone to use compiled in a CRA without ejecting. There is an issue for it here already, though this is not on our radar right now (pull requests are welcome though!): https://github.com/atlassian-labs/compiled/issues/770

Finally, would you be able to post the webpack config you have overridden?

**Reply by onionhammer** [reply]
Would it be a lot of work to put together a CRACO sample so that people can add 'compiled' without ejecting as a workaround for those struggling with this? 

I'm also in a similar situation, trying to avoid ejecting.

**Reply by xorb** [reply]
Same here

**Reply by NikAtIdeally** [reply]
Compiled works really well with CRACO. 
I've used the @compiled/babel-plugin and it worked out of the box.
Only issue I have found so far is is that addComponentName config setting does nothing. Trying to figure out if thats my fault. CRACO's or the babel-plugin bug.
